lib/polyfill/patchedmediakeys_apple.js

```diff
@@ -740,5 +740,5 @@ shaka.polyfill.PatchedMediaKeysApple.MediaKeyStatusMap = class {
   }
 };
 
-
-shaka.polyfill.register(shaka.polyfill.PatchedMediaKeysApple.install);
+// Make it configurable ?
+// shaka.polyfill.register(shaka.polyfill.PatchedMediaKeysApple.install);
```

Comment:
I need the native implementation to work for MSE/EME to work properly with Safari. But we should keep native workflow for people who are not ready to switch to MSE.

Any ideas on how making this polyfill configurable on a configuration basis ?

Reply:
Apple gives guidance on how to use the native implementation with the modern api, wouldn't it make sense to make it work with the modern API?

Look at the zip from https://github.com/google/shaka-player/issues/3346

Reply:
Or uncomment https://github.com/google/shaka-player/blob/master/lib/polyfill/patchedmediakeys_apple.js#L38-L50

Reply:
I've tried to load a TS content manifest and a CMAF content manifest directly as `file` (without MSE) but there aren't any `encrypted` event fired by the browser, therefore no capability to negociate a licence, the video element still send the `webkitneedkey` event. I've tried to replicate as much as possible the Apple examples, but no success at all 🤷‍♂️ Maybe I am missing something

(with the uncomment you mentionned)

Reply:
More info from video-js: https://github.com/videojs/videojs-contrib-eme/commit/58976556e19bd210a075b0190b0d0d45084b98c6 and https://github.com/videojs/videojs-contrib-eme/commits/main